src/main/java/com/matt/forgehax/mods/StepMod.java

```diff
@@ -29,27 +32,89 @@
           .description("how high you can step")
           .defaultTo(1.2f)
           .min(0f)
+          .success(__ -> {
+            MC.addScheduledTask(() -> {
+              if (isEnabled()) {
+                EntityPlayer player = getLocalPlayer();
+                if (player != null) updateStepHeight(player);
+              }
+            });
+          })
+          .build();
+
+  public final Setting<Boolean> unstep =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("unstep")
+          .description("step down instead of falling")
+          .defaultTo(false)
           .build();
 
   public StepMod() {
     super(Category.PLAYER, "Step", false, "Step up blocks");
   }
 
+  @Override
+  protected void onEnabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player != null) wasOnGround = player.onGround;
+  }
+
   @Override
   public void onDisabled() {
-    if (getLocalPlayer() != null) {
-      getLocalPlayer().stepHeight = DEFAULT_STEP_HEIGHT;
+    EntityPlayer player = getLocalPlayer();
+    if (player != null) player.stepHeight = DEFAULT_STEP_HEIGHT;
+  }
+
+  private void updateStepHeight(EntityPlayer player) {
+    if (!MC.isCallingFromMinecraftThread())
```

Comment:
I promise you this isn't necessary. Even if it was, changing player.stepHeight will not cause a concurrent modification exception

Reply:
I have never gotten a concurrent modification exception, even without these checks. I _have_ gotten random and inexplicable errors and bugs caused by race conditions.

This check may not be necessary, but it enforces that you call the function correctly, and in the edge case that a bug slips through the cracks, you'll know exactly what it is.